src/python/ensembl/io/genomio/assembly/status.py

```diff
@@ -0,0 +1,482 @@
+
+    tsv_outfile = f"{output_directory}/{outfile_prefix}.tsv"
+
+    header_list = list(ReportStructure().keys())
+    # header_list.remove("Strain")
+    # header_list.remove("Isolate")
+    header_list = [query_type] + header_list
+
+    with open(tsv_outfile, "w+") as tsv_out:
+        writer = csv.writer(tsv_out, delimiter="\t", lineterminator="\n")
+        writer.writerow(header_list)
+
+        for core, report_meta in parsed_asm_reports.items():
+            final_asm_report = [core] + list(report_meta.values())
+            writer.writerow(final_asm_report)
+        tsv_out.close()
+
+
+# def classify_assembly_status(core_accessions: dict) -> None:
+#     """Main function to pare set of core list and call ncbi datasets"""
+
+
+def main() -> None:
+    """Module's entry-point."""
+    parser = ArgumentParser(
+        description="Track the assembly status of a set of input core(s) using NCBI 'datasets'"
+    )
+    input_group = parser.add_mutually_exclusive_group(required=True)
+    input_group.add_argument(
+        "--input_cores",
+        type=Path,
+        required=False,
+        default=None,
+        help="List of ensembl core database(s) names to retrieve query accessions",
+    )
+    input_group.add_argument(
+        "--input_accessions",
+        type=Path,
+        required=False,
+        default=None,
+        help="List of assembly INSDC query accessions",
+    )
+    parser.add_argument_dst_path(
+        "--download_dir",
+        default="Assembly_report_jsons",
+        help="Folder where the assembly report JSON file(s) are stored",
+    )
+    parser.add_argument_dst_path(
+        "--assembly_report_prefix",
+        default="AssemblyStatusReport",
+        help="Prefix used in assembly report TSV output file.",
+    )
+    parser.add_argument(
+        "--host",
+        type=str,
+        required=False,
+        help="Server hostname (fmt: mysql-ens-XXXXX-YY); required with '--input_cores'",
+    )
+    parser.add_argument(
+        "--port", type=str, required=False, help="Server port (fmt: 1234); required with '--input_cores'"
+    )
+    parser.add_argument(
+        "--datasets_version_url",
+        type=str,
+        required=False,
+        metavar="URL",
+        help="datasets version: E.g. library://lcampbell/ensembl-genomio/ncbi-datasets-v16.10.0:latest",
+    )
+    parser.add_argument(
+        "--cache_dir",
+        type=Path,
+        required=False,
+        default="$NXF_SINGULARITY_CACHEDIR",
+        metavar="SINGULARITY_CACHE",
+        help="Custom path to user generated singularity container housing ncbi tool 'datasets'",
+    )
+    parser.add_argument(
+        "--datasets_batch_size",
+        type=int,
+        required=False,
+        default=100,
+        metavar="BATCH_SIZE",
+        help="Number of accessions requested in one query to datasets",
+    )
+
+    parser.add_log_arguments(add_log_file=True)
+    args = parser.parse_args()
+
+    init_logging_with_args(args)
+
```

Comment:
Just a reminder that most of the following code will need to be moved to its own function if we want to add unit test in the future.